lib/BuildSystem/BuildSystem.cpp

```diff
@@ -524,7 +524,18 @@ class CommandTask : public Task {
   }
 
   virtual void inputsAvailable(BuildEngine& engine) override {
-    command.inputsAvailable(getBuildSystem(engine).getCommandInterface(), this);
+    bool shouldSkip = !getBuildSystem(engine).getCommandInterface()
+      .getDelegate().shouldCommandStart(&command);
+
+    if (shouldSkip) {
+      getBuildSystem(engine).getCommandInterface().taskIsComplete(
+          this, BuildValue::makeSkippedCommand());
+      getBuildSystem(engine).getCommandInterface().getDelegate()
+        .commandFinished(&command);
```

Comment:
This seems weird to me -- we never call `commandStarted` so why call `commandFinished`?